Engine/GameObject.cs

```diff
@@ -1,126 +1,158 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Collections.ObjectModel;
-
 using Engine.BaseAssets.Components;
 
 namespace Engine
 {
-    public sealed class GameObject : SerializedObject
+    public sealed class GameObject : SerializableObject
     {
-        private bool enabled = true;
-        public bool Enabled { get => Transform.Parent == null ? enabled : enabled && Transform.Parent.GameObject.enabled; set => enabled = value; }
-        public Transform Transform { get; private set; }
-        private List<Component> components = new List<Component>();
+        [SerializedField]
+        private string name;
+        [SerializedField]
+        private bool localEnable = true;
+        [SerializedField]
+        private Transform transform = null;
+        [SerializedField]
+        private readonly List<Component> components = new List<Component>();
+
+        public string Name { get => name; set => name = value; }
+        public bool LocalEnabled { get => localEnable; set => localEnable = value; }
+        public Transform Transform { get => transform; private init => transform = value; }
         public ReadOnlyCollection<Component> Components => components.AsReadOnly();
-        internal bool PendingDestroy { get; private set; }
-        public bool Initialized { get; internal set; }
+
+        public bool Enabled => localEnable && Transform != null && (Transform.Parent?.GameObject?.Enabled ?? true); //TODO: maybe better cache value
```

Comment:
use "is not null" instead of !=

Reply:
[https://www.youtube.com/watch?v=o-YBDTqX_ZU&ab_channel=MusRest](Do not check)